### DISCUSSION TITLE
AttributeError: 'NoneType' object has no attribute 'copy'

### DISCUSSION THREAD

**Discussion by celestinoxp** [alan-turing-institute/sktime#6667] [Q&A]
What changed in latest versions to give  AttributeError: 'NoneType' object has no attribute 'copy' ?

More context: i upgrade sktime version in pycaret to support scikit-learn 1.5, then i upgrade sktime from 0.26 to 0.30.... but now tests have  AttributeError: 'NoneType' object has no attribute 'copy'

i can't find the solution... can anyone take a look? https://github.com/pycaret/pycaret/actions/runs/9665451003/job/26662574185?pr=4009

**Comment by fkiraly** [comment]
Hm, nothing I can spot that would cause this failure - it seems `X=None` is arriving at the `_transform`, which should not happen.

There are also no changes in the relevant files that I can spot in the relevant classes - the `BaseTransformer` and the `Imputer`.

I want to note that `sktime 0.30` is not compatible with `sklearn 1.5`, the first version that passes all tests will be `sktime 0.31`, to be released in the next days.

**Reply by celestinoxp** [reply]
@fkiraly I just installed sktime 0.26 even with scikit-learn 1.5 and the tests pass... I will have to check all the changes in sktime up to the most recent version 0.30.x to find out where the change is that generates the error... you can Give me some help to save me time?
Any changes that no longer accept None?

Edited:
version 0.26 ok
version 0.27 fails <<changes here broken pycaret test

Edited2: 
@fkiraly maybe changes here: https://github.com/sktime/sktime/pull/5977 ?

Edited3:
I feel like I don't have enough experience to fix this problem alone... I have doubts whether the test should be modified... the changes in sktime introduce "requires_X" and I don't know if this should be adapted in pycaret...

@fkiraly can you review the test https://github.com/pycaret/pycaret/blob/master/tests/test_time_series_preprocess.py#L319 ?

**Reply by celestinoxp** [reply]
@ngupta23  could easily solve the problem if he's not busy... ping...

**Reply by fkiraly** [reply]
@celestinoxp, I would have to look at pure `sktime` code to start debugging - could you try to isolate the failing code so we have only `sktime` and `sklearn` imports, dummy data, etc?

**Reply by celestinoxp** [reply]
@fkiraly faiklay I'm sorry, I have little experience and yesterday I spent the whole day trying to figure out the problem, and that's why I'm here asking for help, otherwise I would solve it myself... please have some patience and if you can help I would appreciate it

**Reply by celestinoxp** [reply]
I think i found a possible problem here: https://github.com/pycaret/pycaret/blob/master/pycaret/time_series/forecasting/oop.py#L4495, but i can´t change None initial value... i don't k now how to fix it, else sktime gives an errors.. @fkiraly can you take a look?

**Reply by celestinoxp** [reply]
@fkiraly ping...
